### PR TITLE
Allowing torchani to use GPU 

### PR DESCRIPTION
**Pull request by kexul**
## Description
Added config for torchani to use GPU, related to this issue https://github.com/MolSSI/QCEngine/issues/326. 
Now we can use `local_options={'device':'cuda'}` to assign the device: 
```
import qcengine as qcng
import qcelemental as qcel

mol = qcel.models.Molecule.from_data("""
O  0.0  0.000  -0.129
H  0.0 -1.494  1.027
H  0.0  1.494  1.027
""")

inp = qcel.models.AtomicInput(
    molecule=mol,
    driver="energy",
    model={"method": "ani2x"}
    )

ret = qcng.compute(inp, "torchani", local_options={'device':'cuda'})
print(ret.return_result)
```

### CODE REVIEW

**Comment by loriab** [comment]
Thanks for the PR, @kexul . @dotsdl, how does this relate to how OpenMM or OpenFF in general switches cpu/gpu modes?

**Comment by WardLT** [comment]
Instead of having the user manually turn on GPU usage, could we use torch
to detect the presence of a GPU and automatically use it?

**Comment by kexul** [comment]
> Instead of having the user manually turn on GPU usage, could we use torch to detect the presence of a GPU and automatically use it?

That sounds reasonable, I've changed the code according to your suggestion.

File: qcengine/config.py [hunk 1]
```diff
@@ -158,7 +158,7 @@ class TaskConfig(pydantic.BaseModel):
     scratch_messy: bool = pydantic.Field(
         False, description="Leave scratch directory and contents on disk after completion."
     )
-
+
```
Comment (WardLT) [inline qcengine/config.py:161]: Introducing the whitespace here probably caused a problem with our linter.

Could you run `make format` and issue a commit with the changed files?

File: qcengine/programs/torchani.py [hunk 2]
```diff
@@ -172,7 +173,7 @@ def compute(self, input_data: "AtomicInput", config: "TaskConfig") -> "AtomicRes
         ret_data["extras"] = input_data.extras.copy()
         ret_data["extras"].update(
             {
-                "ensemble_energies": energy_array.detach().numpy(),
+                "ensemble_energies": energy_array.cpu().detach().numpy(),
                 "ensemble_energy_avg": energy.item(),
```
Comment (WardLT) [inline qcengine/programs/torchani.py:177]: To make sure I understand correctly, calling `item` will move the "energy" tensor to the cpu? You don't need a `.cpu()` statement, right?

**Review by WardLT — APPROVED** [review]
The changes look great to me. Thanks for the contribution!

Once you fix the linting problems with `make format`, this should be ready to merge.

@loriab , do you want to review this as well? Or, can I click the button?

File: qcengine/programs/torchani.py (hunk 2 above)
Comment (kexul) [reply qcengine/programs/torchani.py:177]: Yes, calling `item()` would move the tensor to CPU, you are absolutely right.

File: qcengine/config.py (hunk 1 above)
Comment (kexul) [reply qcengine/config.py]: Done by https://github.com/MolSSI/QCEngine/pull/327/commits/d7ced823512ce58133739327684cae380592309c  😉

**Comment by loriab** [comment]
Thanks @WardLT , merge at will.

**Pull request merged**